src/compiler/crystal/semantic/call_error.cr

```diff
@@ -282,6 +285,19 @@ class Crystal::Call
     raise message, owner_trace
   end
 
+  def logical_operator?(def_name)
+    return true if convert_def_name_to_logical_operator(def_name)
+    false
+  end
+
+  def convert_def_name_to_logical_operator(def_name)
+    case def_name
+    when "and"; "&&"
+    when "or" ; "||"
```

Comment:
Not sure about the formatting here,  might just do a `Hash(String, String)` to a constant and look up?